Editors/README.md

```diff
@@ -22,56 +22,32 @@ The [Swift for Visual Studio Code extension](https://marketplace.visualstudio.co
 
 ## Sublime Text
 
-Before using SourceKit-LSP with Sublime Text, you will need to install the LSP package from Package Control. To configure SourceKit-LSP, open the LSP package's settings. The following snippet should be enough to get started with Swift.
+Before using SourceKit-LSP with Sublime Text, you will need to install the LSP package from Package Control. And you're ready to go if you have `xcrun` in the `$PATH`.
 
-You will need the path to the `sourcekit-lsp` executable for the "command" section.
+To configure SourceKit-LSP additionally, open the SourceKit-LSP package's settings by typing in command palette `Preferences: LSP-SourceKit Settings`. The following snippet should be enough to get started with Swift and Objective-C/C++ and the custom path to `sourcekit-lsp` executable.
 
 ```json
 {
-  "clients":
-  {
-    "SourceKit-LSP":
-    {
-      "enabled": true,
-      "command": [
+    "command": [
         "<sourcekit-lsp command>"
-      ],
-      "env": {
-        // To override the toolchain, uncomment the following:
-        // "SOURCEKIT_TOOLCHAIN_PATH": "<path to toolchain>",
-      },
-      "languages": [
+    ],
+    "languages": [
         {
-          "scopes": ["source.swift"],
-          "syntaxes": [
-            "Packages/Swift/Syntaxes/Swift.tmLanguage",
-            "Packages/Decent Swift Syntax/Swift.sublime-syntax",
-          ],
```

Comment:
Why don’t we need to specify `syntax` files anymore? sourcekit-lsp only implements semantic highlighting and while it does take that job very far and returns classifications for most tokens, I think it’s good to have an in-process syntactic highlighting as well, eg. when the `textDocument/semanticTokens` requests takes a while to reply.

Reply:
It's because of LSP update itself, not sure for how long, but definitely quite a while LSP plugin relies on sublime text scopes, rather than on selected syntax, e.g. `source.swift`, and semantic highlight feature toggles generally now.. But anyway the current sourcekit-lsp wrapper doesn't support that feature yet. I'll try to update it properly someday I guess.

Here's the references:
1. general LSP related setting [link](https://github.com/sublimelsp/LSP/blob/5891ff41deaa781783effeb97c978b3e50bd0a26/LSP.sublime-settings#L216)
2. kinda golden standard pyright LSP client with the absence of syntaxes settings [link](https://github.com/sublimelsp/LSP-pyright/blob/7a31a89b247b788793e4383e1548345d6f842858/LSP-pyright.sublime-settings#L58)
3. current settings of a sourcekit-lsp client with the very limited setup options [link](https://github.com/sublimelsp/LSP-SourceKit/blob/master/LSP-SourceKit.sublime-settings)

Reply:
Regarding the LSP link: I just want to state that LSP supports semantic highlighting, not syntactic highlighting. Semantic highlighting might take a while (we’re ideally talking 100ms - 1s because we need to build an AST to figure out all the types and whether identifiers are locals, members, system symbols, …) here, my understanding of the LSP spec has always been that clients should have fast syntactic highlighting and enrich that with the semantic highlighting from LSP as it comes in.

But I don’t have enough experience how the current best practices in Sublime-Land are and will leave the decision up to you.

Reply:
Yep, this is exactly the case. The syntactic highlighting handled by a `*.sublime-syntax` files powered by a st engine. But the catch is that those settings are completely unrelated to lsp functioning.

The reason why those settings were here before was that it was the way how the lsp plug-in mapped which server to call on a given file, but isn't true now.

Reply:
Btw, speaking about syntactic highlighting, those that were here were really limited in their implementation, finally there is a way better one now implemented. [this one](https://forums.swift.org/t/swift-next-swift-syntax-package-in-async-await-era-for-sublime-text-4/58227). So just in case, if you had an idea where to put a highlight of it to make swift st users life a little less painful, I'd really appreciate that.

Reply:
Oh, that’s great. I have personally been waiting for some syntactic highlighting package that contains recent Swift syntax for Sublime. Thanks 🙏🏽 

How about you mention it in this document?

Reply:
Ah, you just can't imagine how I was thrilled when found it occasionally. 

Done. Looking forward to it being merged, if there're no issues left.

Reply:
Ah, the last one, FYI: not working semantic syntax highlighting for a swift was a Sublime LSP plugin side bug, and it's almost [fixed](https://github.com/sublimelsp/LSP/pull/2453) now.

Reply:
`languages` and `languageId` is deprecated in ST LSP. Instead `"selector": "source.swift | source.c"' (etc.) should be used.

ST3 doesn't support `selector` but ST3 has not been relevant for many years now. ST4 is what matters.